include/battle/ai_context.h

```diff
@@ -0,0 +1,53 @@
+#ifndef POKEPLATINUM_BATTLE_AI_CONTEXT_H
+#define POKEPLATINUM_BATTLE_AI_CONTEXT_H
+
+#include "constants/battle.h"
+#include "constants/moves.h"
+
+#include "struct_defs/struct_020790C4.h"
+#include "struct_decls/struct_itemdata_decl.h"
+
+typedef struct AIContext {
+    u8 scriptCursor;
+    u8 moveSlot;
+    u16 moveNum;
+
+    s8 moveCurrPP[4];
+
+    int calcTemp;
+    u32 thinkingMask;
+
+    u8 stateFlags;
+    u8 thinkingBitShift;
+    u8 padding0012;
+    u8 padding0013;
+    int padding0014;    // this does not match with a u32, weird
+
+    u8 moveDamageRolls[LEARNED_MOVES_MAX];
+    
+    u16 battlerMoves[MAX_BATTLERS][LEARNED_MOVES_MAX];
+    u8 battlerAbilities[MAX_BATTLERS];
+    u16 battlerHeldItems[MAX_BATTLERS];
+
+    u16 trainerItems[MAX_OPPONENTS][MAX_TRAINER_ITEMS];
+
+    u32 scriptStackPointer[8];
+    u8 scriptStackSize;
+
+    u8 trainerItemCounts[MAX_BATTLERS_PER_SIDE];
+    u8 battlerIdAttacker;
+    u8 battlerIdDefender;
+    u8 usedItemType[MAX_BATTLERS_PER_SIDE];
+    u8 usedItemCondition[MAX_BATTLERS_PER_SIDE];
+    u8 padding0082;     // originally implicit, declaring here as explicit
+
+    u16 usedItemNums[MAX_BATTLERS_PER_SIDE];
```

Comment:
This name makes no sense, each battler has only one entry, how can there be multiple ItemNums for them?
Confused as to how this is used.

Reply:
It's definitely one item per battler; my bad on keeping this one plural to make that confusing.